### PR TITLE
Remove remaining references to CMYK

### PR DESCRIPTION
**Pull request by rlafuente**
any modern printing pipeline handles the RGB-CMYK conversion directly, and handling of CMYK at the application level is a pretty outdated practice.

in any case our CMYK support is pretty bad, so it should just go.

### CODE REVIEW

File: shoebot/data/basecolor.py [hunk 1]
```diff
@@ -87,7 +84,7 @@ def __init__(self, *args, **kwargs):
                     a / color_range,
                 )
 
-        # Three to five parameters, either RGB, RGBA, HSB, HSBA, CMYK, CMYKA
+        # Three to five parameters, either RGB, RGBA, HSB, HSBA,
```
Comment (stuaxo) [inline shoebot/data/basecolor.py:87]: Trailing comma.

**Comment by stuaxo** [comment]
Does this break our copy of the nodebox color lib ?

**Comment by rlafuente** [comment]
haven't tested it, but can do it tonight -- just to be sure, you mean the external colors lib?

**Comment by stuaxo** [comment]
I think we included in our codebase under lib?

**Comment by karstenw** [comment]
...
> in any case our CMYK support is pretty bad, so it should just go.

That doesnt stop the world from having CMYK images.

If the support is bad it should get better, not disappear.

**Comment by stuaxo** [comment]
I'm more on the side of keeping it in, a big OTOH is that the SB codebase is a mess and making it small is a positive.

**Comment by stuaxo** [comment]
One big issue I had originally, when porting tests over was having no idea what was correct or not.

**Comment by rlafuente** [comment]
> That doesnt stop the world from having CMYK images.
> 
> If the support is bad it should get better, not disappear.

I agree with you! But "bad" here means it was hastily implemented, never tested (or even used), and always converted to RGB for the Cairo graphics backend, since it does not support CMYK. I suppose Cocoa's tighter control on color management at the OS level made things simpler for Nodebox and Drawbot.

From my knowledge, it's pretty useless (at least for print purposes) to use CMYK colors which will be converted back into RGB. At most there's educational value -- which in any case shouldn't be discounted, and I'd be happy to support CMYK in the future once we review the way we deal with colors (offloading some logic to libs like [colormath](https://python-colormath.readthedocs.io/)).

But for now, we're trying hard to finish or remove half-working things, and CMYK support needs thought on the best way to implement; while we're not there, I think it's best not to pretend that we support it when we know it's very half-baked.

If you have a different perspective on this, I'm all ears :)

**Comment by karstenw** [comment]
...
> I agree with you! But "bad" here means it was hastily implemented, never tested (or even used), and always converted to RGB for the Cairo graphics backend, since it does not support CMYK. I suppose Cocoa's tighter control on color management at the OS level made things simpler for Nodebox and Drawbot.

Nodebox and Drawbot use the OS for cmyk handling.

...
> But for now, we're trying hard to finish or remove half-working things, and CMYK support needs thought on the best way to implement; while we're not there, I think it's best not to pretend that we support it when we know it's very half-baked.
> 
> If you have a different perspective on this, I'm all ears :)

A few years back I wrote a python/bottle based image database for a small publisher. To create the thumbnails for many thousand CMYK files (mostly eps and tif) I had to script Photoshop which boils down to GUI scripting, which is awful, because I couldnt find a Python lib that could handle CMYK images with a clipping path or alpha channel.

That's the thing most image tools need: an easy way to create a RGBA representation of a CMYK file.

What should you do? I don't know. If current CMYK support sux say so in the docs. Everybody will know it's suboptimal but perhaps one day someone is willing to invest the 20-30 hours it needs to make it better. That person will make faster progress, if the current stuff is findable (search "cyan" etc.)

**Comment by karstenw** [comment]
What I'm trying to say: we should open the door for cmyk images (by converting them to RGBA) but we don't need to work in cmyk. The support must not include saving as cmyk. That is the playgound for the photoshops.

**Comment by rlafuente** [comment]
Ah, we're very much in sync then -- the changes in this PR are about removing internal CMYK color representations (mostly in color() and Color objects), and don't touch image imports.

For your purposes, we should look into how Cairo handles incoming CMYK images. I'm making an issue so that we can track this -- #389 

So from your comment, I suppose it is okay to merge this PR?

(Unrelated, but i'm curious about your python/bottle pipelines for image handling, since i've done some of that professionally too. Do you have any examples of that online?)

**Comment by karstenw** [comment]
> Ah, we're very much in sync then -- the changes in this PR are about removing internal CMYK color representations (mostly in color() and Color objects), and don't touch image imports.
> 
> For your purposes, we should look into how Cairo handles incoming CMYK images. I'm making an issue so that we can track this -- #389

I was looking at shoebot/data/img.py Image.init() which uses `img = PILImage.open(path)`
That should be enough for now.

> So from your comment, I suppose it is okay to merge this PR?

Yes.

> (Unrelated, but i'm curious about your python/bottle pipelines for image handling, since i've done some of that professionally too. Do you have any examples of that online?)

- bottle/web is for queries and meta data editing (which images of an article number go to the webshop, which is main image,...)

- an external programm (python & pyobjc) does the database updates. It scans the images folder and updates/deletes/changes image records, uses photoshop to create at least 6 RGB thumbnails (1 original size and 4 size for the shop). Thumbs are stored in a hash-directory:

screen 2021-04-23 at 02 15 41

- The component after the '@' is the hor/vert maxsize.
- The user installs a schema handler program that allows him to
  - open the original file in photoshop
  - place the file in Quark or indesign
  - open it's containing folder
- All by just clicking a link

**Comment by rlafuente** [comment]
Hey, inspiring :) I miss doing custom script work for print! I'm sure you had a reason not to use Imagemagick if you went all the way to the GUI scripting route...

Anyway it feels a bit awkward to progress a conversation on a PR that's about to close... if you feel like it, we're hanging around at the [#shoebot matrix channel](https://matrix.to/#/#shoebot:matrix.org) (you can access from a browser via [element.io](https://element.io/)). If it's not your thing, let's keep ping-ponging over here, it's encouraging to have you around!

**Pull request merged**